### ISSUE TITLE
can't create static ipsets when starting befw-firewalld

### ISSUE THREAD

**Issue by al-joshua** [wgnet/befw#9]
Hi there,

I'm playing with your tool and while running through the demo I receive following error. Could you please help me with that?

```
agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$  befw-firewalld -noroot -debug -config befw.conf
2020/12/28 13:14:30 Opening queue
2020/12/28 13:14:30 Unbinding to selected family
2020/12/28 13:14:30 nflog_unbind_pf failed
2020/12/28 13:14:30 Binding to selected family
2020/12/28 13:14:30 Creating queue
2020/12/28 13:14:30 nflog_bind_group failed
NFLogger started, you can get information from /var/run/befw/*
[API]: Can't run local socket:  listen unix /var/run/befw/api.sock: bind: address already in use
[Consist] can't create static ipsets
[Rules] IPSet refresh error: 
-------
create rules_allow hash:net
create tmp-wODtb1Rj7sOBmGq0NRTr134JX hash:net
flush tmp-wODtb1Rj7sOBmGq0NRTr134JX
swap tmp-wODtb1Rj7sOBmGq0NRTr134JX rules_allow
destroy tmp-wODtb1Rj7sOBmGq0NRTr134JX

-------
Error while creating ipset rules_allow 
Can't apply state:  
Error while refresh():  ; sleeping for  40  seconds
^C2020/12/28 13:14:38 Closing queue

```

**Comment by annmuor** [comment]
Sure. You're running it with insufficient privileges. Any ipset stuff requires either root or net_admin cap.

**Comment by al-joshua** [comment]
Hey, thanks for prompt reply! but I can't run it with sudo
`agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$ sudo befw-firewalld -debug -config befw.conf
sudo: befw-firewalld: command not found`

sorry if that looks stupid..

**Comment by al-joshua** [comment]
I also get this:

```
agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$ sudo systemctl status befw-firewalld.service 
● befw-firewalld.service - Backend Firewall Service
   Loaded: loaded (/usr/lib/systemd/system/befw-firewalld.service; disabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2020-12-28 15:34:50 MSK; 1s ago
  Process: 2043 ExecStart=/usr/bin/befw-firewalld (code=exited, status=203/EXEC)
 Main PID: 2043 (code=exited, status=203/EXEC)

Dec 28 15:34:50 net-hbfw01 systemd[1]: befw-firewalld.service: Main process exited, code=exited, status=203/EXEC
Dec 28 15:34:50 net-hbfw01 systemd[1]: befw-firewalld.service: Failed with result 'exit-code'.
Dec 28 15:34:50 net-hbfw01 systemd[1]: befw-firewalld.service: Service hold-off time over, scheduling restart.
Dec 28 15:34:50 net-hbfw01 systemd[1]: befw-firewalld.service: Scheduled restart job, restart counter is at 5.
Dec 28 15:34:50 net-hbfw01 systemd[1]: Stopped Backend Firewall Service.
Dec 28 15:34:50 net-hbfw01 systemd[1]: befw-firewalld.service: Start request repeated too quickly.
Dec 28 15:34:50 net-hbfw01 systemd[1]: befw-firewalld.service: Failed with result 'exit-code'.
Dec 28 15:34:50 net-hbfw01 systemd[1]: Failed to start Backend Firewall Service.

```

**Comment by annmuor** [comment]
It's a question of your PATH environment settings.

**Comment by al-joshua** [comment]
I guess I've managed to start with sufficient privillege but the error is the same

`agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$ sudo -E env "PATH=$PATH" befw-firewalld -debug -config befw.conf
2020/12/28 15:55:03 Opening queue
2020/12/28 15:55:03 Unbinding to selected family
2020/12/28 15:55:03 Binding to selected family
2020/12/28 15:55:03 Creating queue
NFLogger started, you can get information from /var/run/befw/*
2020/12/28 15:55:03 Try Run
[Consist] can't create static ipsets
[Rules] IPSet refresh error: 
-------
create rules_allow hash:net
create tmp-wODtb1Rj7sOBmGq0NRTr134JX hash:net
flush tmp-wODtb1Rj7sOBmGq0NRTr134JX
swap tmp-wODtb1Rj7sOBmGq0NRTr134JX rules_allow
destroy tmp-wODtb1Rj7sOBmGq0NRTr134JX

-------
Error while creating ipset rules_allow 
Can't apply state:  
Error while refresh():  ; sleeping for  40  seconds
[Consist] can't create static ipsets
[Rules] IPSet refresh error: 
-------
`

**Comment by annmuor** [comment]
Do you have ipset & iptables binaries in your path?
try sudo ipset list

**Comment by al-joshua** [comment]
this looks fine

```
agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$ env | grep "PATH"
GOPATH=/home/agorovoy/go
MANPATH=:/opt/puppetlabs/puppet/share/man
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/puppetlabs/bin:/usr/local/go/bin:/home/agorovoy/go/bin
agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$ sudo ipset list
agorovoy@net-hbfw01:~/go/src/github.com/wgnet/befw/demo$ 

```

**Comment by al-joshua** [comment]
I've reinstalled everything under root user, but still got the same issue. Although I can create ipset manually, this does not help. Would be grateful for any thoughts here.

```
root@net-hbfw01:~/go/src/github.com/wgnet/befw/demo# /root/go/bin/befw-firewalld -debug -config befw.conf
2020/12/28 18:19:47 Opening queue
2020/12/28 18:19:47 Unbinding to selected family
2020/12/28 18:19:47 Binding to selected family
2020/12/28 18:19:47 Creating queue
[ INFO ] [befw-firewalld/befw-firewalld.go:53] NFLogger started, you can get information from /var/run/befw/*
[ WARN ] 2020/12/28 18:19:47 Try Run
[rules.go:318] [Consist] can't create static ipsets
[ WARN ] [rules.go:230] [Rules] IPSet refresh error:
[ DEBUG ] [rules.go:231] create rules_allow hash:net
create tmp-wODtb1Rj7sOBmGq0NRTr134JX hash:net
flush tmp-wODtb1Rj7sOBmGq0NRTr134JX
swap tmp-wODtb1Rj7sOBmGq0NRTr134JX rules_allow
destroy tmp-wODtb1Rj7sOBmGq0NRTr134JX

[ WARN ] [state.go:269] Error while creating ipsetrules_allow
[ WARN ] [state.go:312] Can't apply state: 
[ WARN ] [run.go:53] Error while refresh():; sleeping for 40 seconds

```

```
root@net-hbfw01:~/go/src/github.com/wgnet/befw/demo# ipset list
root@net-hbfw01:~/go/src/github.com/wgnet/befw/demo# 
root@net-hbfw01:~/go/src/github.com/wgnet/befw/demo# ipset create rules_allow hash:net
root@net-hbfw01:~/go/src/github.com/wgnet/befw/demo# ipset list
Name: rules_allow
Type: hash:net
Revision: 6
Header: family inet hashsize 1024 maxelem 65536
Size in memory: 344
References: 0
Number of entries: 0
Members:


```

**Comment by annmuor** [comment]
check if ipset is present at /usr/sbin/ipset. if not, create a symlink there.
I'll fix that later.

**Comment by al-joshua** [comment]
hey, mate, that workaround worked out! Similar issue with iptables-restore. Keep in mind

```
agorovoy@net-hbfw01:/usr/sbin$ ll | grep iptables-restore
lrwxrwxrwx  1 root root      22 Dec 28 20:52 iptables-restore -> /sbin/iptables-restore*
```

**Comment by annmuor** [comment]
Fixed in ed4fcf03cb40a4ff83386357b172fcd6e228d178
